src/ADAL.PCL.Desktop/CryptographyHelper.cs

```diff
@@ -72,20 +79,20 @@ public byte[] SignWithCertificate(string message, X509Certificate2 certificate)
                     // For .NET 4.6 and below we get the old RSACryptoServiceProvider implementation as the default.
                     // Try and get an instance of RSACryptoServiceProvider which supports SHA256
                     newRsa = GetCryptoProviderForSha256(cspRsa);
+
+                    using (var sha = new SHA256Cng())
+                    {
+                        return newRsa.SignData(messageBytes, sha);
+                    }
                 }
                 else
                 {
-                    // For .NET Framework 4.7 and onwards the RSACng implementation is the default.
-                    // Since we're targeting .NET Framework 4.5, we cannot actually use this type as it was
-                    // only introduced with .NET Framework 4.6.
-                    // Instead we try and create an RSACryptoServiceProvider based on the private key from the
-                    // certificate.
-                    newRsa = GetCryptoProviderForSha256(certificate);
-                }
-
-                using (var sha = new SHA256Cng())
-                {
-                    return newRsa.SignData(messageBytes, sha);
+                    CngKey key = GetCngPrivateKey(certificate);
+                    using (RSACng rsaCng = new RSACng(key))
```

Comment:
The doc https://msdn.microsoft.com/en-us/library/dn957360(v=vs.110).aspx says that if the key is null, the constructor of RSACng will throw an exception. I'm not clear what happens when the user is not allowed to access the certificate, @kpanwar  (the full trust assertion). Will this provoke an exception which will be bubbled-up from this method?